### PR TITLE
feat: Support multiple DNS providers in cert-manager

### PR DESCRIPTION
**Pull request by tumido**
Resolves: https://github.com/operate-first/apps/issues/1119
Related: https://github.com/operate-first/SRE/issues/374

I've created a new IAM user in AWS with the suggested role:
https://cert-manager.io/docs/configuration/acme/dns01/route53/#set-up-an-iam-role

This PR also enables cert manager on Rick (since we already had an overlay in cert-manager folder yet it was not installed) and Balrog (the operator was installed but not configured at all)

### CODE REVIEW

**Comment by tumido** [comment]
CI fails with:

```
Error: accumulating resources: accumulation err='accumulating resources from '../common': '/home/prow/go/src/github.com/operate-first/apps/cert-manager/overlays/route53/eu-west-2/common' must resolve to a file': recursed accumulation of path '/home/prow/go/src/github.com/operate-first/apps/cert-manager/overlays/route53/eu-west-2/common': replace operation does not apply: doc is missing path: /spec/acme/solvers/0/dns01/route53/region: missing value
Aborting due to errexit on line 43. Exit code: 1
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2021-10-05T11:20:34Z"}
```

This is because CI masks the ksops secret generator - hence the overlay using the [openshift-ingress-issuer_patch.enc.yaml](https://github.com/operate-first/apps/blob/538bbcbe5e8c540cf14385cc88d6875c588c832d/cert-manager/overlays/route53/common/openshift-ingress-issuer_patch.enc.yaml) file fails in CI. Confirmed that it works locally, please verify that it works for you as well and override the CI check once confirmed, please. :slightly_smiling_face:

File: cert-manager/base/openshift-ingress-issuer.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: ingress-letsencrypt-production
+  namespace: openshift-ingress
+spec:
+  acme:
```
Comment (HumairAK) [inline cert-manager/base/openshift-ingress-issuer.yaml:7]: when I build the balrog ovelray, the `email` `server` and `privateKeySecretRef` don't seem to get consumed, I get: 

```yaml
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: ingress-letsencrypt-production
  namespace: openshift-ingress
spec:
  acme:
    solvers:
    - dns01:
        cnameStrategy: Follow
        route53:
          accessKeyID: <redacted>
          region: eu-west-2
          secretAccessKeySecretRef:
            key: secret-access-key
            name: route53
      selector:
        dnsZones:
        - aws.operate-first.cloud


```

Comment (tumido) [reply cert-manager/base/openshift-ingress-issuer.yaml:7]: You're right, seems the ksops "merge" annotation doesn't work as expected :slightly_frowning_face:  I'll see what I can do.

Comment (tumido) [reply cert-manager/base/openshift-ingress-issuer.yaml:7]: I've removed it from ksops. we expose the access key id now. :shrug:

**Pull request closed without merging**